modules/exploits/multi/http/atutor_sqli.rb

```diff
@@ -234,7 +234,7 @@ def service_details
   def exploit
     print_status("Dumping the username and password hash...")
     credz = dump_the_hash
-    if credz
+    if credz && credz != []
```

Comment:
```suggestion
    if credz && !credz.empty?
```

Also, given that there's no point continuing if this conditional fails, the module could `return` here.

```ruby
return unless credz
return if credz.empty?
```

However, the module does not `print_` anywhere else, and it only uses `fail_with` in the methods used later (`login` and `upload_shell`)... so, in the event that dumping the hash fails, the module will effectively just die with no feedback after `Dumping the username and password hash...`

A more user friendly approach would be:

```ruby
if credz.nil? || credz.empty?
  fail_with(NotVulnerable, 'Failed to retrieve username and password hash')
end
```

Reply:
I like these suggestions, thanks for the input.  I'll implement in a few days